dao/postgres/entities/pg-subscriptions.js

```diff
@@ -98,20 +98,58 @@ class PgSubscriptions {
             return callback(null, subsList);
         });
     }
-
+  
     getAllImpl(filter, orderBy, offset, limit, noCountCache, callback) {
         debug(`getAll(filter: ${filter}, orderBy: ${orderBy}, offset: ${offset}, limit: ${limit})`);
+        //return callback(new Error('PG.getAllImpl: Not implemented.'));
         const fields = [];
         const values = [];
         const operators = [];
-        this.pgUtils.addFilterOptions(filter, fields, values, operators);
+        const joinedFields = [
+           {
+                source: 'a.api_group',
+                as: 'api_group',
+                alias: 'apiGroup'
+            },       
+            {
+                source: 'a.data->>\'approved\'',
+                as: 'approved',
+                alias: 'approved'
+            }, 
+            {
+                source: 'b.application_name',
+                as: 'application_name',
+                alias: 'applicationName'
+            },
+            {
+                source: 'b.owner',
+                as: 'owner',
+                alias: 'owner'
+            },
+            {
+                source: 'b.user',
+                as: 'user',
+                alias: 'user'
+            },
+            {
+                source: 'b.userid',
+                as: 'userid',
+                alias: 'userid'
+            }
+     
+        ];
+        this.pgUtils.addFilterOptions(filter, fields, values, operators, joinedFields);
+        // This may be one of the most complicated queries we have here...
         const options = {
             limit: limit,
             offset: offset,
             orderBy: orderBy ? orderBy : 'id ASC',
             operators: operators,
             noCountCache: noCountCache,
+            joinedFields: joinedFields,
+            joinClause: 'INNER JOIN (SELECT string_agg(o.data->>\'email\', \', \') as owner, string_agg(r.name, \', \') as user, string_agg(r.users_id, \', \') as userid, p.data->> \'name\' as application_name , p.id FROM wicked.applications p, wicked.owners o, wicked.registrations r WHERE o.applications_id = p.id AND o.users_id = r.users_id GROUP BY application_name, p.id) b ON b.id = a.applications_id'
```

Comment:
Not super happy with this approach, which is concatenating owner names and emails into single fields, correct? The API is usually complete JSON based, but here we break that. How difficult is it to attempt to convert this to a proper JSON structure?

I understand it may be really challenging, but please give it a thought or two?

Reply:
You mean concat it post retrieval? we will loose filtering and sorting in that case no?

Reply:
No, concat to a JSON array somehow.

Reply:
Ideas could include to_json and array_agg operators in Postgres.

Reply:
```
select s.*, a.*, (
	select to_json(array_agg(row_to_json(tmp))) from (
		select o.applications_id as applications_id, o.users_id as users_id, o.data->>'role' as role, o.data->>'email' as email, r.name from wicked.owners o inner join wicked.registrations r on r.users_id = o.users_id where r.pool_id = 'wicked'
	) tmp
	where tmp.applications_id = a.id) as owner_data
from wicked.subscriptions s
inner join wicked.applications a on a.id = s.applications_id
```

Reply:
Only issue with above approach is that I cannot make a WHERE clause from owner_data column. I had to re-fire the subquery to make sure that owner email and owner user name end up in the columns so as to make the filtering and ordering works. Or at least I can't find more better way

Reply:
fixed but with my ugly hack